plugins/modules/vmware_host_iscsi.py

```diff
@@ -0,0 +1,874 @@
+
+                # add port binds in an iSCSI configuration
+                if self.update_port_bind_flag:
+                    for vnic in list(filter(lambda x: x not in self.existing_system_iscsi_config['port_bind'],
+                                            self.port_bind)):
+                        try:
+                            self.host_obj.configManager.iscsiManager.BindVnic(iScsiHbaName=self.vmhba_name,
+                                                                              vnicDevice=vnic)
+                            result['changed'] = True
+                        except Exception as e:
+                            self.module.fail_json(msg="Failed to add a port bind: %s" % to_native(e))
+
+        if self.state == 'absent':
+            self.check_hba_name()
+            self.diff_iscsi_config()
+
+            if self.module.check_mode:
+                self.module.exit_json(changed=self.change_flag)
+
+            if self.change_flag:
+                self.generate_iscsi_config()
+
+                # remove a dynamic target to an iSCSI configuration
+                if self.remove_send_interface_flag:
+                    try:
+                        self.host_obj.configManager.storageSystem.RemoveInternetScsiSendTargets(
+                            iScsiHbaDevice=self.vmhba_name,
+                            targets=self.send_target_configs)
+                        result['changed'] = True
+                    except Exception as e:
+                        self.module.fail_json(msg="Failed to remove a dynamic target: %s" % to_native(e))
+
+                # remove a static target to an iSCSI configuration
+                if self.remove_static_interface_flag:
+                    try:
+                        self.host_obj.configManager.storageSystem.RemoveInternetScsiStaticTargets(
+                            iScsiHbaDevice=self.vmhba_name,
+                            targets=self.static_target_configs)
+                        result['changed'] = True
+                    except Exception as e:
+                        self.module.fail_json(msg="Failed to remove a static target: %s" % to_native(e))
+
+                # remove port binds from an iSCSI configuration
+                if self.remove_port_bind_flag:
+                    for vnic in list(set(self.existing_system_iscsi_config['port_bind']) & set(self.port_bind)):
+                        try:
+                            self.host_obj.configManager.iscsiManager.UnbindVnic(iScsiHbaName=self.vmhba_name,
+                                                                                vnicDevice=vnic,
+                                                                                force=self.force)
+                            result['changed'] = True
+                        except Exception as e:
+                            self.module.fail_json(msg="Failed to remove a port bind: %s" % to_native(e))
+
+        if result['changed'] is True:
```

Comment:
It could be very handy to get the config returned even there was no change (except if that's a bad practice in modules).

Taking my test scenario, I use first `vmware_host_iscsi` to enable the software adapter. From the register of that tasks, I get the `vmhba_name` to call a second time the `vmware_host_iscsi` module to set the Dynamic target for instance (see playbook snip here under). But this will fail in case the iSCSI Software adapter is already enabled as the `iscsi_properties` are not returned (see the outputs here under).

An other possibility could be to provide a module like `vmware_hosts_iscsi_facts`. (Note that I saw in the integration tests that you get the `vmhba_name` but I don't find that handy at all)

### Snip of my test playbook
```
    - name: Enable iSCSI Software adapter on esxi nodes
      vmware_host_iscsi:
        hostname: "{{ ansible_host }}"
        username: "{{ ansible_user }}"
        password: "{{ ansible_ssh_pass }}"
        esxi_hostname: "{{ ansible_host }}"
        validate_certs: no
        state: enabled
      register: node_iscsi_adapt
      delegate_to: localhost

    - name: Print iscsi adapter
      debug:
        var: node_iscsi_adapt

    - name: Configure iSCSI dynamic targets of esxi nodes
      vmware_host_iscsi:
        hostname: "{{ ansible_host }}"
        username: "{{ ansible_user }}"
        password: "{{ ansible_ssh_pass }}"
        esxi_hostname: "{{ ansible_host }}"
        validate_certs: no
        state: present
        iscsi_config:
          vmhba_name: "{{ node_iscsi_adapt.iscsi_properties.vmhba_name }}"
          send_target:
            - address: "{{ nimble_iscsi_ip1 }}"
            - address: "{{ nimble_iscsi_ip2 }}"
      register: cnode_iscsi_config
      delegate_to: localhost
```

### Output on change
```
TASK [Enable iSCSI Software adapter on nodes] ***********************************************************************************************************************************************************************************************
Tuesday 16 June 2020  11:46:59 +0200 (0:00:02.176)       0:00:02.324 ********** 
changed: [node03 -> localhost]

TASK [Print iscsi adapter] *******************************************************************************************************************************************************************************************************************
Tuesday 16 June 2020  11:47:03 +0200 (0:00:03.968)       0:00:06.292 ********** 
ok: [node03] => 
  node_iscsi_adapt:
    changed: true
    failed: false
    iscsi_properties:
      iscsi_alias: ''
      iscsi_authentication_properties:
        _vimtype: vim.host.InternetScsiHba.AuthenticationProperties
        chapAuthEnabled: false
        chapAuthenticationType: chapProhibited
        chapInherited: null
        chapName: ''
        chapSecret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
        mutualChapAuthenticationType: chapProhibited
        mutualChapInherited: null
        mutualChapName: ''
        mutualChapSecret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
      iscsi_enabled: true
      iscsi_name: iqn.1998-01.com.vmware:node03-02f56186
      iscsi_send_targets: []
      iscsi_static_targets: []
      port_bind: []
      vmhba_name: vmhba64
```

### Output on status 'ok'
```
TASK [Enable iSCSI Software adapter on nodes] ***********************************************************************************************************************************************************************************************
Tuesday 16 June 2020  11:46:08 +0200 (0:00:00.194)       0:00:00.194 **********
ok: [node01 -> localhost]

TASK [Print iscsi adapter] *******************************************************************************************************************************************************************************************************************
Tuesday 16 June 2020  11:46:11 +0200 (0:00:03.803)       0:00:03.998 **********
ok: [node01] =>
  node_iscsi_adapt:
    changed: false
    failed: false
```

Reply:
okay :)
I'll be creating a vmware_hosts_iscsi_info module (It may take some time, but ...)